app/jobs/geocode_location_job.rb

```diff
@@ -47,6 +51,6 @@ def enqueue_next_job
   end
 
   def next_location
-    Location.where('latitude IS NULL OR longitude IS NULL').first
+    Location.where(geocode_response: nil).first
```

Comment:
I thought I'd missed something! I see your point about creating a custom error and then not throwing or catching it. I could use a string or another value instead. What do you think?